doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+.. _ug_matter_device_memory_profiling:
+
+Profiling memory in Matter applications
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide offers useful commands and Kconfig options for configuring memory sizes for stacks, heaps, and Zephyr settings.
+You can learn to measure memory usage and adjust these values according to your project's requirements.
+
+Measuring memory usage
+======================
+
+You can use several predefined Kconfig options and UART shell commands to obtain the current memory statistics from the device.
+To do this, you can enable the global Matter memory profiling Kconfig, which also activates all other necessary Kconfig options.
+Alternatively, you can enable each option separately.
+
+The Matter memory profiling Kconfig option is a predefined setting that enables all UART shell commands for measuring memory usage.
+To enable it, set the :kconfig:option:`CONFIG_CHIP_MEMORY_PROFILING` Kconfig option to ``y``.
+
+This Kconfig option enables the following functionalities on the Matter device:
+
+- :ref:`ug_matter_configuring_settings_shell` by setting the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
+- :doc:`matter:nrfconnect_examples_cli` by setting the :kconfig:option:`CONFIG_CHIP_LIB_SHELL` Kconfig option to ``y``.
+- Zephyr Kernel commands by setting the :kconfig:option:`CONFIG_KERNEL_SHELL` Kconfig option to ``y``.
+- Zephyr Settings shell by setting the :kconfig:option:`CONFIG_SETTINGS_SHELL` Kconfig option to ``y``.
+- OpenThread shell by setting the :kconfig:option:`CONFIG_OPENTHREAD_SHELL` Kconfig option to ``y`` if you build the Matter over Thread variant.
+- Chip statistics shell by setting the :kconfig:option:`CONFIG_CHIP_STATISTICS` Kconfig option to ``y``.
+
+All the functionalities listed below are automatically enabled if the Matter memory profiling Kconfig option is activated.
+However, you can also find the specific Kconfig options required for each functionality to enable them separately.
+
+Heap usage and Matter-related statistics
+----------------------------------------
+
+Requirements:
+
+.. parsed-literal::
+   :class: highlight
+
+    CONFIG_SHELL=y
+    CONFIG_CHIP_STATISTICS=y
+    CONFIG_CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT=y
+    CONFIG_CHIP_MALLOC_SYS_HEAP=y
+    CONFIG_SYS_HEAP_RUNTIME_STATS=y
+
+You can measure heap usage by monitoring peak usage.
+To do this, you should first reset the current peak measurement, then read the peak usage, perform some operations on the device, and read the peak usage again.
+he difference between the initial and subsequent peak values indicates the number of bytes dynamically allocated by all operations performed by the device during that interval.
```

Comment:
It isn't that easy, right? :smile: I think we should mention it may be measured this way, but it will not always work, as some objects may be also destroyed in the meantime, so the peak value will not change even if we allocated several things. 
```suggestion
The difference between the initial and subsequent peak values indicates the number of bytes dynamically allocated by all operations performed by the device during that interval.
```